### PR TITLE
Shipping Labels M1: analytics

### LINKED ISSUE #2930: Shipping Labels M1 release: add tracking events
As per discussion in `p91TBi-3oZ`, we need to add the following events to the Shipping Labels M1 feature:

| Event | Event props | Description |
| ------------- | ------------- | ------------- |
| `*_shipping_label_refund_requested `  | N/A  |  triggered when a refund request is successfully submitted from the app i.e. refund API is called from the app  |
| `*_shipping_label_print_requested `  | N/A  |  triggered when users select a paper size and initiate the printing process. i.e. print shipping label API is called from the app  |
| `*_shipment_tracking_menu_action`  | `action=track`  |  triggered when users clicks on the track shipment menu button from the app  |
| `*_shipment_tracking_menu_action`  | `action=copy`  |  triggered when users clicks on the copy shipment menu button from the app  |
| `*_shipping_label_api_request`  | `action={success, failed}`  |  triggered when a GET shipping labels API request is submitted from the app. The event property `action` can be `success` = if the API response is successful; `failed` = if the API request failed  |

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #2930 

## Why

This PR tracks events for shipping labels M1!

## Changes

- Added `WooAnalyticsStat` cases for new shipping labels M1 events
- Added two `WooAnalyticsEvent` functions for events with properties: `shipmentTrackingMenuAction` and `shippingLabelsAPIRequest`
- Implemented event tracking in corresponding places, and added test cases when possible

## Testing

- Go to the orders tab
- Tap on an order with at least a non-refunded shipping label —> after syncing, should see `🔵 Tracked shipping_label_api_request, properties: [AnyHashable("action"): "success”, …]` in the console and a shipping label package card should appear
- In “Tracking number” row in the shipping label package card, tap the ellipsis menu
- Tap “Copy tracking number” in the action sheet or popover —> should see `🔵 Tracked shipment_tracking_menu_action, properties: [AnyHashable("action"): "copy”, …]` in the console
- In “Tracking number” row in the shipping label package card, tap the ellipsis menu again
- Tap “Track shipment” in the action sheet or popover —> should see `🔵 Tracked shipment_tracking_menu_action, properties: [AnyHashable("action"): "track”, …]` in the console
- Tap “Reprint Shipping Label”
- Tap “Print Shipping Label” —> should see `🔵 Tracked shipping_label_print_requested` in the console
- Tap the ellipsis menu at the top of the shipping label package card (to the right of “Package #”)
- Tap “Request a Refund”
- Tap “Refund Label ($#)” —> should see `🔵 Tracked shipping_label_refund_requested` in the console
- Go offline
- Go back to the orders tab and tap on the same order —> after syncing, should see `🔵 Tracked shipping_label_api_request, properties: [AnyHashable("action"): “failed”, …]` in the console and a shipping label package card should appear

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift [hunk 1]
```diff
@@ -86,6 +98,14 @@ extension WooAnalyticsEvent {
     static func featureFeedbackBanner(context: FeedbackContext, action: FeatureFeedbackBannerAction) -> WooAnalyticsEvent {
         WooAnalyticsEvent(statName: .featureFeedbackBanner, properties: ["context": context.rawValue, "action": action.rawValue])
     }
+
+    static func shipmentTrackingMenu(action: ShipmentTrackingMenuAction) -> WooAnalyticsEvent {
+        WooAnalyticsEvent(statName: .shipmentTrackingMenuAction, properties: ["action": action.rawValue])
+    }
+
+    static func shippingLabelsAPIRequest(result: ShippingLabelsAPIRequestResult) -> WooAnalyticsEvent {
+        WooAnalyticsEvent(statName: .shippingLabelsAPIRequest, properties: ["action": result.rawValue])
```
Comment (jaclync) [inline WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift:107]: I proposed to rename `action` to `result` in p91TBi-3oZ-p2#comment-2769 - will update when we have a decision!

**Review by ctarda — APPROVED** [review]
I can see all the events mentioned in the PR description being triggered. Here are some screenshots:

Screenshot 2020-12-28 at 5 29 05 AM
Screenshot 2020-12-28 at 5 28 35 AM
Screenshot 2020-12-28 at 5 27 14 AM

:shipit:

File: WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift:107]: Per answer in p91TBi-3oZ-p2#comment-2777, we're keeping the `action` key and I also updated the event name to be consistent with Android in: https://github.com/woocommerce/woocommerce-ios/pull/3410/commits/a2af63732eaf755ec0f925de8c652729bfdefb79

**Comment by jaclync** [comment]
Thank you for reviewing over the holidays César 💟 merging now!

**Pull request merged**